ovos_tts_plugin_piper/__init__.py

```diff
@@ -123,60 +123,112 @@ class PiperTTSPlugin(TTS):
         'upc_pau-x-low': 'https://github.com/rhasspy/piper/releases/download/v0.0.2/voice-ca-upc_pau-x-low.tar.gz',
         'vos-x-low': 'https://github.com/rhasspy/piper/releases/download/v0.0.2/voice-vi-vivos-x-low.tar.gz'
     }
+    engines = {}
 
     def __init__(self, lang="en-us", config=None):
         super(PiperTTSPlugin, self).__init__(lang, config)
-        model = self.config.get("model") or \
-                self.lang2voices.get(lang) or \
-                self.lang2voices.get(lang.split("-"[0]))
-        if isinstance(model, list):
-            model = model[0]
-
-        if not os.path.isfile(model) and model in self.voice2url:
-            xdg_p = f"{xdg_data_home()}/piper_tts/{model}"
-            if not os.path.isdir(xdg_p):
-                model = self.voice2url[model]
-
-        if model.startswith("http"):
-            m = model.split("/")[-1]
-            xdg_p = f"{xdg_data_home()}/piper_tts/{m.split('.')[0]}"
-
-            model_tar = f"{xdg_p}/{m}"
-            if not os.path.isfile(model_tar):
-                LOG.info(f"downloading piper model: {model}")
-                os.makedirs(xdg_p, exist_ok=True)
-                data = requests.get(model)
-                with open(model_tar, "wb") as f:
-                    f.write(data.content)
-
-                with tarfile.open(model_tar) as file:
-                    file.extractall(xdg_p)
-            for f in os.listdir(xdg_p):
-                if f.endswith(".onnx"):
-                    model = f"{xdg_p}/{f}"
-                    LOG.debug(f"selected model: {model}")
-                    break
+        if self.voice == "default":
+            if self.lang.startswith("en"):
```

Comment:
I don't like this. Does this not limit all English models to Alan pope?

Reply:
this is only if voice is not set, if you request a voice in your config this is never triggered

Reply:
Yeah, maybe it should take the first one *(or random pick)* from the `lang`, for example with French:

```json
        'fr': ['gilles-low', 'mls_1840-low', 'siwis-low', 'siwis-medium'],
```

First one will be `gilles-low` but random could be any of the four.

Reply:
random pick is bad, since every request becomes a different voice...

the first is  always picked, this logic is just merging en-gb to en-us ....

is american and british english a different language? :)

Reply:
Yes, my wife has a friend in the uk and there are several small differences. Mainly things that would effect stt  more that tts. "Lift" vs "elevator",  "flat" vs "apartment "

Reply:
well, mycroft has been en-us + alan pope since the early days.... this is just keeping it that way